code/modules/assembly/infrared.dm

```diff
@@ -141,70 +171,143 @@
 		return
 
 	if(href_list["state"])
-		on = !(on)
-		update_icon()
+		activate()
 
 	if(href_list["visible"])
 		visible = !(visible)
 
 		if(beam)
 			beam.set_visible(visible)
+		update_icon()
 
 	if(href_list["close"])
 		usr << browse(null, "window=infra")
 		return
 
+	if(href_list["direction"])
+		var/choice = input("What direction will you aim the laser toward?","Infrared Laser") as null|anything in list("NORTH", "EAST", "SOUTH", "WEST")
+		if (choice)
+			dir = text2dir(choice)
+			update_icon()
+			if (beam)
+				returnToPool(beam)
+				beam = null
+			process()
+
 	if(usr)
 		attack_self(usr)
 
-	return
-
-
-/obj/item/device/assembly/infra/verb/rotate()//This could likely be better
-	set name = "Rotate Infrared Laser"
-	set category = "Object"
-	set src in usr
-
-	dir = turn(dir, 90)
-	return
 
 /***************************IBeam*********************************/
 
 /obj/effect/beam/infrared
 	name = "i beam"
-	icon = 'icons/obj/projectiles.dmi'
-	icon_state = "ibeam"
+	icon = 'icons/effects/beam.dmi'
+	icon_state = "infrared"
 	var/limit = null
-	var/visible = 0.0
+	var/visible = TRUE
 	var/left = null
-	anchored = 1.0
+	anchored = TRUE
 	flags = 0
 
-
 	var/obj/item/device/assembly/infra/assembly
+	var/puffed = 0
+
+	var/list/smokes_n_mists = list(
+		/obj/effect/decal/chemical_puff,
+		/obj/effect/effect/smoke,
+		/obj/effect/effect/water,
+		/obj/effect/effect/foam,
+		/obj/effect/effect/steam,
+		/obj/effect/mist,
+		)
+
+/obj/effect/beam/infrared/Destroy()
+	assembly = null
+	..()
 
-/obj/effect/beam/infrared/proc/hit()
-	if(assembly)
-		assembly.trigger_beam()
+/obj/effect/beam/infrared/resetVariables()
+	..("sources", "children", "limit","visible", "left", "assembly", "invisibility", "puffed", args)
+	children = list()
+	sources = list()
+
+/obj/effect/beam/infrared/get_damage()
+	return 0
+
+/obj/effect/beam/infrared/update_icon()
+	puffed = 0
+	if (!master)
+		invisibility = 101
+	else if (visible)
+		invisibility = 0
```

Comment:
There's no define for this, sadly :pensive:

Reply:
😔